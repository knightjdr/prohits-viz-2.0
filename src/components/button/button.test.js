import React from 'react';
import { shallow } from 'enzyme';

import Button from './button';

const handleClick = jest.fn();

describe('Button component', () => {
  describe('with default props', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Button
          onClick={handleClick}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have default class', () => {
      expect(wrapper.hasClass('button_default')).toBeTruthy();
    });

    it('should call click function', () => {
      wrapper.simulate('click');
      expect(handleClick).toHaveBeenCalled();
    });
  });

  describe('with user props', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Button
          className="test-class"
          onClick={handleClick}
          miscProp="something"
          type="warning"
        />,
      );
    });

    it('should have added class', () => {
      expect(wrapper.hasClass('test-class')).toBeTruthy();
    });

    it('should have added prop', () => {
      expect(wrapper.prop('miscProp')).toBe('something');
    });

    it('should have button type as class', () => {
      expect(wrapper.hasClass('button_warning')).toBeTruthy();
    });
  });
});
