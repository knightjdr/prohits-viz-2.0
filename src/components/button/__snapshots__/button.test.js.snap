// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component with default props should match snapshot 1`] = `
<button
  className="button_default"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;
